### PR TITLE
[Form] Add `duplicate_preferred_choices` option to `ChoiceType`

### LINKED ISSUE #35135: [Form] Allow disabling of preferred_choices being shown twice
Although I can see the reason behind this change in Symfony 4.4 (see https://symfony.com/blog/new-in-symfony-4-4-form-improvements#show-preferred-choices-twice ), I'd like to be able to disable this - i.e. I'd like to have an option to display the preferred_choices just once (on top of the list, as in previous versions).

The option could be called `preferred_choices_twice` (with `true` as default).

### PR DESCRIPTION
**Pull request by arnaud-deabreu**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #35135 
| License       | MIT
| Doc PR        | TODO

I've reintroduced layout tests as they were before #32658, is that enough? /cc @xabbuh :)
Thanks @HeahDude for your mentoring!

### CODE REVIEW

File: src/Symfony/Component/Form/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ CHANGELOG
    `model_timezone` option in `DateType`, `DateTimeType`, and `TimeType`
  * Deprecate `PostSetDataEvent::setData()`, use `PreSetDataEvent::setData()` instead
  * Deprecate `PostSubmitEvent::setData()`, use `PreSubmitDataEvent::setData()` or `SubmitDataEvent::setData()` instead
+ * Add `duplicate_preferred_choices` option in `ChoiceType`
+ * Add `$duplicatePreferredChoices` parameter to `ChoiceListFactoryInterface::createView`
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/CHANGELOG.md]:
```suggestion
 * Add `$duplicatePreferredChoices` parameter to `ChoiceListFactoryInterface::createView()`
```

File: src/Symfony/Component/Form/ChoiceList/Factory/CachingFactoryDecorator.php [hunk 2]
```diff
@@ -145,8 +145,9 @@ public function createListFromLoader(ChoiceLoaderInterface $loader, mixed $value
         return $this->lists[$hash];
     }
 
-    public function createView(ChoiceListInterface $list, mixed $preferredChoices = null, mixed $label = null, mixed $index = null, mixed $groupBy = null, mixed $attr = null, mixed $labelTranslationParameters = []): ChoiceListView
+    public function createView(ChoiceListInterface $list, mixed $preferredChoices = null, mixed $label = null, mixed $index = null, mixed $groupBy = null, mixed $attr = null, mixed $labelTranslationParameters = []/* , bool $duplicatePreferredChoices = true */): ChoiceListView
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/ChoiceList/Factory/CachingFactoryDecorator.php:151]:
```suggestion
    /**
     * @param bool $duplicatePreferredChoices
     */
    public function createView(ChoiceListInterface $list, mixed $preferredChoices = null, mixed $label = null, mixed $index = null, mixed $groupBy = null, mixed $attr = null, mixed $labelTranslationParameters = []/* , bool $duplicatePreferredChoices = true */): ChoiceListView
```

please add similar annotations to announce the new argument everywhere it's added (public methods only of course)

**Review by nicolas-grekas — COMMENTED** [review]
Thanks for taking care of this, @javiereguiluz should like it :)

File: src/Symfony/Component/Form/ChoiceList/Factory/CachingFactoryDecorator.php (hunk 2 above)
Comment (arnaud-deabreu) [reply src/Symfony/Component/Form/ChoiceList/Factory/CachingFactoryDecorator.php:151]: Thanks for your review!
I added the annotations and added some comments to describe the new agument in the interface.

**Review by nicolas-grekas — APPROVED** [review]
(failures unrelated)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @arnaud-deabreu.

**Pull request merged**